packages/strapi-plugin-content-manager/admin/src/components/DynamicZone/CategoryItem.js

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Collapse } from 'reactstrap';
+import Banner from './Banner';
+import DynamicComponentCard from '../DynamicComponentCard';
+
+const CategoryItem = ({
+  category,
+  components,
+  isOpen,
+  isFirst,
+  onClickToggle,
+  onClickComponent,
+}) => {
+  const [showComponents, setShowComponents] = useState(false);
+  useEffect(() => {
+    if (isOpen) {
+      setShowComponents(true);
+    }
+  }, [isOpen]);
+
+  return (
+    <>
+      <Banner onClickToggle={onClickToggle} isFirst={isFirst} isOpen={isOpen} category={category} />
+      <Collapse
+        isOpen={isOpen}
+        style={{ backgroundColor: '#F3F3F3' }}
+        onExited={() => setShowComponents(false)}
+      >
+        {showComponents && (
+          <div
+            className="componentsList"
+            style={{ paddingTop: '10px', paddingLeft: '15px', paddingRight: '15px' }}
+          >
```

Comment:
Can you create a file for this component please.

`...src/components/DynamicZone/ComponentsList.js``

```js
import styled from 'styled-components';

const ComponentsList = styled.div`
  padding-top: 10px;
  padding-left: 15px;
  padding-right: 15px;
`;

export default ComponentsList
```

and then in `.../src/components/DynamicZone/CategoryItem.js`

```javascript
...
return (
    <>
      <Banner onClickToggle={onClickToggle} isFirst={isFirst} isOpen={isOpen} category={category} />
      <Collapse isOpen={isOpen} onExited={handleExited}>
        {showComponents && (
          <ComponentsList className="componentsList">
            {components.map(component => {
              const {
                info: { icon, name: friendlyName },
                componentUid,
              } = component;

              return (
                <DynamicComponentCard
                  key={componentUid}
                  componentUid={componentUid}
                  friendlyName={friendlyName}
                  icon={icon}
                  onClick={() => onClickComponent(componentUid)}
                />
              );
            })}
          </ComponentsList>
        )}
      </Collapse>
    </>
  );
...
```

Reply:
@HichamELBSI done in 5963e6c46e7d9323a211a75212db05524964d926